src/ComputeSharp.Graphics/Buffers/ConstantBuffer{T}.cs

```diff
@@ -44,31 +44,20 @@ private static int GetPaddedSize()
         public T this[int i] => throw new InvalidExecutionContextException($"{nameof(ConstantBuffer<T>)}<T>[int]");
 
         /// <inheritdoc/>
-        public override void GetData(Span<T> span, int offset, int count)
+        public override unsafe void GetData(Span<T> span, int offset, int count)
         {
             if (IsPaddingPresent)
             {
-                // Create the temporary array
-                byte[] temporaryArray = ArrayPool<byte>.Shared.Rent(count * PaddedElementSizeInBytes);
-                Span<byte> temporarySpan = temporaryArray.AsSpan(0, count * PaddedElementSizeInBytes);
-
                 // Copy the padded data to the temporary array
                 using (MappedResource resource = MapResource())
                 {
-                    MemoryHelper.Copy(resource.Pointer, offset * PaddedElementSizeInBytes, temporarySpan, 0, count * PaddedElementSizeInBytes);
-                }
-
-                ref byte tin = ref temporarySpan.GetPinnableReference();
-                ref T tout = ref span.GetPinnableReference();
-
-                // Copy the padded data to the target span, removing the padding
-                for (int i = 0; i < count; i++)
-                {
-                    ref byte rsource = ref Unsafe.Add(ref tin, i * PaddedElementSizeInBytes);
-                    Unsafe.Add(ref tout, i) = Unsafe.As<byte, T>(ref rsource);
+                    ref var dest = ref MemoryMarshal.GetReference(span);
+                    byte* untypedSrc = (byte*)resource.Pointer;
```

Comment:
General code style take, not really a big fan of raw pointers here as I find they make the code less readable. Couldn't we keep using `ref` values as before, since the performance should be the same anyway?